semgrep-core/parsing/Parse_csharp_tree_sitter.ml

```diff
@@ -1368,7 +1368,7 @@ and statement (env : env) (x : CST.statement) =
        let v3 = expression env v3 in
        let v4 = token env v4 (* ")" *) in
        let v5 = statement env v5 in
-       todo env (v1, v2, v3, v4, v5)
+       OtherStmt (OS_Sync, [E v3; S v5])
```

Comment:
Actually for those kind of stmts that have inside other statements, it's better to use the
OtherStmtWithStmt construct. It's better than OtherStmt because when we build
a control-flow graph for a generic AST (see pfff/lang-GENERIC/analyze/IL_to_CFG.ml),
we need to know those nested stmts,